tests/build/multi-platform.go

```diff
@@ -304,6 +449,165 @@ func getAwsInstances() ([]string, error) {
 	return ret, nil
 }
 
+func createConfigMapForHostPool(f *framework.Framework) error {
+	armInstances, err := getHostPoolAwsInstances()
+	if err != nil {
+		return fmt.Errorf("error getting aws host pool instances: %v", err)
+	}
+	hostConfig := &v1.ConfigMap{}
+	hostConfig.Name = HostConfig
+	hostConfig.Namespace = ControllerNamespace
+	hostConfig.Labels = map[string]string{MultiPlatformConfigKey: "hosts"}
+
+	hostConfig.Data = map[string]string{}
+	count := 0
+	for _, instance := range armInstances {
+		hostConfig.Data[fmt.Sprintf("host.aws-arm64-%d.address", count)] = instance
+		hostConfig.Data[fmt.Sprintf("host.aws-arm64-%d.platform", count)] = AwsPlatform
+		hostConfig.Data[fmt.Sprintf("host.aws-arm64-%d.user", count)] = Ec2User
+		hostConfig.Data[fmt.Sprintf("host.aws-arm64-%d.secret", count)] = AwsSecretName
+		hostConfig.Data[fmt.Sprintf("host.aws-arm64-%d.concurrency", count)] = "4"
+		count++
+	}
+
+	_, err = f.AsKubeAdmin.CommonController.CreateConfigMap(hostConfig, ControllerNamespace)
+	if err != nil {
+		return fmt.Errorf("error creating host-pool config map: %v", err)
+	}
+	return nil
+}
+
+func createConfigMapForDynamicInstance(f *framework.Framework) error {
+	hostConfig := &v1.ConfigMap{}
+	hostConfig.Name = HostConfig
+	hostConfig.Namespace = ControllerNamespace
+	hostConfig.Labels = map[string]string{MultiPlatformConfigKey: "hosts"}
+
+	hostConfig.Data = map[string]string{}
+	hostConfig.Data["dynamic-platforms"] = AwsPlatform
+	hostConfig.Data["instance-tag"] = DynamicInstanceTag
```

Comment:
This must by dynamically generated, or else concurrent tests will see each others instances as being active and only a single instance will exist at a time (or else max-instances should be higher than 1).

We count the instances using the AWS APIs, we don't maintain a local count on the cluster.

Reply:
IIUC correctly, we have to run the host-pool and dynamic allocation tests one after the other serially, not parallel since both is for plaform type `arm64` and also we have to update the config map as well in between the tests and restart the pod. So I don't think keeping tag fixed for dynamic allocation and different than what is used in host-pool allocation tests is a problem for now. Both will not interfere with each other.

Tough we have a cleanup step to terminate the instance in the `AfterAll` section of the test, assuming due to some unforeseeable reason if the instance is not cleaned up, so I kept the max-instances to 1, so that subsequent test should fail and instances are not leaked anymore. WDYT?